pkg/api/types.go

```diff
@@ -1827,3 +1827,34 @@ func AddToNodeAddresses(addresses *[]NodeAddress, addAddresses ...NodeAddress) {
 		}
 	}
 }
+
+// Type and constants for component health validation.
+type ComponentConditionType string
+
+// These are the valid conditions for the component.
+const (
+	ComponentHealthy ComponentConditionType = "Healthy"
+)
+
+type ComponentCondition struct {
+	Type    ComponentConditionType `json:"type" description:"the type of condition"`
+	Status  ConditionStatus        `json:"status" description:"the status of this condition"`
```

Comment:
This file shouldn't have description tags. Description tags need to be on all non-inline fields in the versioned types.go files.